testsuite/MDAnalysisTests/analysis/test_pca.py

```diff
@@ -145,6 +151,113 @@ def test_transform_universe():
+    group = u.select_atoms('resnum 1')
+    with pytest.raises(ValueError) as exc:
+        func = pca.project_single_frame(0, group=group, anchor=None)
+    assert ("'anchor' cannot be 'None'" +
+            " if 'group' is not 'None'") in str(exc.value)
+
+
+def test_project_more_anchor(u, pca):
+    group = u.select_atoms('resnum 1')
+    with pytest.raises(ValueError) as exc:
+        project = pca.project_single_frame(0, group=group, anchor='backbone')
+    assert "More than one 'anchor' found in residues" in str(exc.value)
+
+
+def test_project_less_anchor(u, pca):
+    group = u.select_atoms('all')
+    with pytest.raises(ValueError) as exc:
+        project = pca.project_single_frame(0, group=group, anchor='name CB')
+    assert ("Some residues in 'group'" +
+            " do not have an 'anchor'") in str(exc.value)
+
+
+def test_project_invalid_anchor(u):
+    pca = PCA(u, select='name CA').run()
+    group = u.select_atoms('all')
+    with pytest.raises(ValueError) as exc:
+        project = pca.project_single_frame(0, group=group, anchor='name N')
+    assert "Some 'anchors' are not part of PCA class" in str(exc.value)
+
+
+@pytest.mark.xfail
+def test_project_compare_projections(u_project):
+    # projections along different PCs should be different
+    pca = PCA(u_project, select=SELECTION).run()
+    project0 = pca.project_single_frame(0)
+    project1 = pca.project_single_frame(1)
+
+    u_project.trajectory[0]
+    coord0 = project0(u_project.trajectory.ts).positions
+    u_project.trajectory[0]
+    coord1 = project1(u_project.trajectory.ts).positions
+    assert_almost_equal(coord0, coord1, decimal=5)
+
+
+def test_project_reconstruct_whole(u, u_project):
+    # structure projected along all PCs
+    # should be same as the original structure
+    pca = PCA(u_project, select=SELECTION).run()
+    project = pca.project_single_frame()
+
+    coord_original = u.trajectory.ts.positions
+    coord_reconstructed = project(u_project.trajectory.ts).positions
+    assert_almost_equal(coord_original, coord_reconstructed, decimal=5)
+
+
+@pytest.mark.parametrize(
+    ("n1", "n2"),
+    [(0, 0), (1, 1), ([0, 1], [0, 1]),
+     pytest.param(0, 1, marks=pytest.mark.xfail),
+     pytest.param(1, 0, marks=pytest.mark.xfail)]
+)
+def test_project_twice_projection(u_project, n1, n2):
+    # Two succesive projections are applied. The second projection does nothing
+    # if both projections are along the same PC(s).
+    pca = PCA(u_project, select=SELECTION).run()
+
+    project_first = pca.project_single_frame(n1)
+    project_second = pca.project_single_frame(n2)
+
+    u_project.trajectory[0]
+    coord1 = project_first(u_project.trajectory.ts).positions.copy()
+    coord2 = project_second(u_project.trajectory.ts).positions.copy()
+    assert_almost_equal(coord1, coord2, decimal=5)
+
+
+@pytest.mark.parametrize(
+    "sel",
+    ['resnum 1 and name CA CB CG',
+     pytest.param('resnum 1 and name N CB CG', marks=pytest.mark.xfail)]
```

Comment:
Is this something that should eventually work? Then leave it here with xfail.

However, this is one of the cases that you catch with a ValueError. Therefore, add a separate test `test_project_ValueError_toomany_anchors` or something like that. Importantly, the test should be testing (and making it clear) that it wants to see an exception because you're constructing the failure case.

Reply:
No, this should always fail.
The purpose is to show that anchor atom (and only the anchor atom) indeed works like an anchor. Removed the second test case though since the errors will be caught by other tests anyway.